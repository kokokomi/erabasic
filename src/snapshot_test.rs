use std::fs;
use pest::Parser;
use crate::{EraBasicParser, Rule};
use insta::assert_debug_snapshot;

#[test]
fn functions() {
    let unparsed_file = fs::read_to_string("example/functions.erb").expect("cannot read file");
    let file = EraBasicParser::parse(Rule::file, &unparsed_file)
        .expect("unsuccessful parse");
    assert_debug_snapshot!(file);
}

#[test]
fn if_statement() {
    let unparsed_file = fs::read_to_string("example/if_statement.erb").expect("cannot read file");
    let file = EraBasicParser::parse(Rule::file, &unparsed_file)
        .expect("unsuccessful parse");
        
    assert_debug_snapshot!(file);
}

#[test]
fn sif_statement() {
    let unparsed_file = fs::read_to_string("example/sif_statement.erb").expect("cannot read file");
    let file = EraBasicParser::parse(Rule::file, &unparsed_file)
        .expect("unsuccessful parse");

    assert_debug_snapshot!(file);
}

#[test]
fn repeat_statement() {
    let unparsed_file = fs::read_to_string("example/repeat_statement.erb").expect("cannot read file");
    let file = EraBasicParser::parse(Rule::file, &unparsed_file)
        .expect("unsuccessful parse");

    assert_debug_snapshot!(file);
}

#[test]
fn while_statement() {
    let unparsed_file = fs::read_to_string("example/while_statement.erb").expect("cannot read file");
    let file = EraBasicParser::parse(Rule::file, &unparsed_file)
        .expect("unsuccessful parse");

    assert_debug_snapshot!(file);
}

#[test]
fn do_loop_statement() {
    let unparsed_file = fs::read_to_string("example/do_loop_statement.erb").expect("cannot read file");
    let file = EraBasicParser::parse(Rule::file, &unparsed_file)
        .expect("unsuccessful parse");

    assert_debug_snapshot!(file);
}