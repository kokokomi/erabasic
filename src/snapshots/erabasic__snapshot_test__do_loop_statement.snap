---
source: src/snapshot_test.rs
expression: file
---
[
    Pair {
        rule: function_declare,
        span: Span {
            str: "@do_loop_statement\n  do\n    ; block\n  loop 1\n",
            start: 0,
            end: 45,
        },
        inner: [
            Pair {
                rule: function_identifier,
                span: Span {
                    str: "do_loop_statement",
                    start: 1,
                    end: 18,
                },
                inner: [
                    Pair {
                        rule: identifier,
                        span: Span {
                            str: "do_loop_statement",
                            start: 1,
                            end: 18,
                        },
                        inner: [],
                    },
                ],
            },
            Pair {
                rule: function_body,
                span: Span {
                    str: "do\n    ; block\n  loop 1\n",
                    start: 21,
                    end: 45,
                },
                inner: [
                    Pair {
                        rule: statement,
                        span: Span {
                            str: "do\n    ; block\n  loop 1",
                            start: 21,
                            end: 44,
                        },
                        inner: [
                            Pair {
                                rule: do_loop_statement,
                                span: Span {
                                    str: "do\n    ; block\n  loop 1",
                                    start: 21,
                                    end: 44,
                                },
                                inner: [
                                    Pair {
                                        rule: do_keyword,
                                        span: Span {
                                            str: "do",
                                            start: 21,
                                            end: 23,
                                        },
                                        inner: [],
                                    },
                                    Pair {
                                        rule: COMMENT,
                                        span: Span {
                                            str: "; block",
                                            start: 28,
                                            end: 35,
                                        },
                                        inner: [],
                                    },
                                    Pair {
                                        rule: block,
                                        span: Span {
                                            str: "\n",
                                            start: 35,
                                            end: 36,
                                        },
                                        inner: [],
                                    },
                                    Pair {
                                        rule: loop_keyword,
                                        span: Span {
                                            str: "loop",
                                            start: 38,
                                            end: 42,
                                        },
                                        inner: [],
                                    },
                                    Pair {
                                        rule: expression,
                                        span: Span {
                                            str: "1",
                                            start: 43,
                                            end: 44,
                                        },
                                        inner: [
                                            Pair {
                                                rule: primary,
                                                span: Span {
                                                    str: "1",
                                                    start: 43,
                                                    end: 44,
                                                },
                                                inner: [
                                                    Pair {
                                                        rule: integer_literal,
                                                        span: Span {
                                                            str: "1",
                                                            start: 43,
                                                            end: 44,
                                                        },
                                                        inner: [],
                                                    },
                                                ],
                                            },
                                        ],
                                    },
                                ],
                            },
                        ],
                    },
                ],
            },
        ],
    },
    Pair {
        rule: EOI,
        span: Span {
            str: "",
            start: 45,
            end: 45,
        },
        inner: [],
    },
]
